### PR TITLE
Refactor/magic improvements

### PR DESCRIPTION
**Pull request by HP4k1h5**

### CODE REVIEW

File: assets/css/components/modal.scss [hunk 1]
```diff
@@ -2,6 +2,7 @@
   display: flex;
   flex-direction: column;
   overflow-x: auto;
```
Comment (jnschrag) [inline assets/css/components/modal.scss:4]: This can be simplified to `overflow: auto`.

File: components/visualizer/MagicChart.vue [hunk 2]
```diff
@@ -1,5 +1,8 @@
 <template>
-  <Modal @close="updateMagicChart({ showMagicChart: false })">
+  <Modal
+    class="magic-chart modal"
```
Comment (jnschrag) [inline components/visualizer/MagicChart.vue:3]: We need to modify this a bit. The `.modal` class is being applied to two elements, the containing div and the `role="dialog"` div, which opens this up to conflict with each other. I think we can probably remove `modal` from the `<Modal>` tag, but take a look and let me know what you think.

File: assets/css/components/magic-chart.scss [hunk 3]
```diff
@@ -1,8 +1,8 @@
 @import '~/assets/css/components/highcharts-theme';
 
-.magic-chart {
-  min-width: 45vw;
-  min-height: 45vh;
+.magic-chart .modal {
+  width: 45vw;
```
Comment (jnschrag) [inline assets/css/components/magic-chart.scss:4]: Can we try this instead?
```
width: 100%;
max-width: 680px;
max-height: 85vh;
margin: rem(16);
```

Using `vw` for the width units here causes it to shrink a little oddly as the screen gets smaller.

**Review by jnschrag — CHANGES_REQUESTED** [review]

File: assets/css/components/magic-chart.scss (hunk 3 above)
Comment (HP4k1h5) [reply assets/css/components/magic-chart.scss]: had a tough time getting the right behavior with these. setting a width was shifting the modal all the way left in the screen

**Review by jnschrag — APPROVED** [review]

**Pull request merged**